critiquebrainz/frontend/views/user.py

```diff
@@ -12,26 +12,64 @@
 from critiquebrainz.frontend import flash
 from critiquebrainz.frontend.forms.log import AdminActionForm
 from critiquebrainz.frontend.login import admin_view
-
+import uuid
 user_bp = Blueprint('user', __name__)
 
 
-@user_bp.route('/<uuid:user_id>')
-def reviews(user_id):
-    user_id = str(user_id)
-    if current_user.is_authenticated and current_user.id == user_id:
-        user = current_user
-    else:
+def get_user_from_username_or_id(username_or_id):
+    """
+    Get user by username or id.
+    """
+    try:
```

Comment:
We can simplify this by doing it in 1 SQL query. For instance in `get_by_id`, we have
```sql
SELECT {columns}
  FROM "user"
 WHERE id = :user_id
```
This can be changed to 
```sql
SELECT {columns}
  FROM "user"
 WHERE musicbrainz_id = :user_ref OR id::text = :user_ref
```
Further, you can add `COALESCE(musicbrainz_id, id::text) AS user_ref` to `USER_GET_COLUMNS` and update `User` class to get the ref from this. This will ensure that the dict and the class always have the same content.

Also, I think instead also renaming the function `get_user_by_id` to `get_user_by_ref`, you cam create a new `get_user_by_ref`. I assume `get_user_by_id` touches too many unrelated places in the codebase. We can migrate all uses to a single db function in a separate PR.

Reply:
Correct. I've now added this function, and updated database queries to fetch user_ref directly from the database instead of computing it manually at a couple of places.